packages/apps/frontera/src/routes/organizations/src/components/shared/RangeSelectorFilter/RangeSelectorFilter.tsx

```diff
@@ -21,96 +22,127 @@ const formatNumberWithCommas = (value: string | number | undefined): string => {
   return numString.replace(/\B(?=(\d{3})+(?!\d))/g, ',');
 };
 
-export const RangeSelector = ({
-  filter,
-  placeholder,
-  onChange,
-  property,
-  years = false,
-  ...rest
-}: RangeSelectorProps) => {
-  const store = useStore();
-  const getWorkFlow = store.workFlows
-    .toArray()
-    .filter((wf) => toJS(wf.value.type === WorkflowType.IdealCustomerProfile));
-  const getWorkFlowId = getWorkFlow.map((wf) => wf.value.id);
+export const RangeSelector = observer(
```

Comment:
This just turned from being shared and abstract into store bound. Not sure it's the right place for it or it impacts other places where it's used.